Components/Model/Product.php

```diff
@@ -240,7 +240,21 @@ public function loadData(Article $article, Shop $shop = null)
         }
         $this->url = $this->assembleProductUrl($article, $shop);
         $this->name = $article->getName();
-        $this->imageUrl = ImageHelper::assembleImageUrl($article, $shop);
+
+        $imageUrls = ImageHelper::getImageUrls($article, $shop);
+        if ($imageUrls) {
```

Comment:
This logic should be moved to image helper. Basically we would have two methods in image helper_ `getMainImage` and 'getAlternativeImages'. Main image would return the string and the other one either array or collection